openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ReturnTypeGenerator.java

```diff
@@ -210,89 +231,64 @@ private ReturnTypeDescriptorNode getReturnNodeForStatusCode200WithContent(
 
         int contentTypeNumber = contentEntries.size();
         if (contentTypeNumber > 1) {
-            Optional<TypeDescriptorNode> unionNode = handleMultipleContents(contentEntries);
-            TypeDescriptorNode type =
-                    unionNode.orElseGet(() -> createSimpleNameReferenceNode(createIdentifierToken(HTTP_RESPONSE)));
+            TypeDescriptorNode type = handleMultipleContents(contentEntries);
             returnNode = createReturnTypeDescriptorNode(returnKeyWord, createEmptyNodeList(), type);
         } else {
             for (Map.Entry<String, MediaType> next : contentEntries) {
-                Optional<TypeDescriptorNode> mediaTypeToken = getMediaTypeToken(next);
-                if (mediaTypeToken.isEmpty()) {
-                    BuiltinSimpleNameReferenceNode type = createBuiltinSimpleNameReferenceNode(null,
-                            createIdentifierToken(ANYDATA));
+                String recordName = getNewRecordName();
+                ImmutablePair<Optional<TypeDescriptorNode>, Optional<TypeDefinitionNode>>
+                        mediaTypeToken = handleMediaType(next, recordName);
+                // right node represents the newly generated node for if there is an inline record in the returned
+                // tuple.
+                Optional<TypeDefinitionNode> rightNode = mediaTypeToken.right;
+
+                if (rightNode.isPresent()) {
+                    typeInclusionRecords.put(recordName, rightNode.get());
+                    setCountForRecord(countForRecord + 1);
```

Comment:
since the variable is within the class we can simply increment like this. 
```suggestion
                    countForRecord++;
```